apps/aecore/src/aec_mining.erl

```diff
@@ -89,18 +94,13 @@ create_coinbase_tx() ->
             Error
     end.
 
--spec adjust_target(block()) -> block().
-adjust_target(Block) ->
+-spec adjust_target(block(), list(header())) -> block().
+adjust_target(Block, AdjHeaders) ->
     Header = aec_blocks:to_header(Block),
-    case aec_target:determine_delta_header_height(Header) of
-        {ok, InitialHeaderHeight} ->
-            %% For mining we can grab the header at height=DeltaHeight from
-            %% the current main chain, hence aec_conductor:get_header_by_height/1 call.
-            %% TODO: It is not good to call back to aec_conductor
-            %%       but we do this for now.
-            {ok, InitialHeader} = aec_conductor:get_header_by_height(InitialHeaderHeight),
-            CalculatedTarget = aec_target:recalculate(Header, InitialHeader),
+    case aec_governance:blocks_to_check_difficulty_count() of
+        N when N == length(AdjHeaders) ->
```

Comment:
It looks like this shall hardcode the expectation that if block height is greater than `aec_governance:blocks_to_check_difficulty_count()` then AdjHeaders *must* have the correct lenght.

It appears to me that not having this check may cause off-by-one somewhere in the code not to adjust target.